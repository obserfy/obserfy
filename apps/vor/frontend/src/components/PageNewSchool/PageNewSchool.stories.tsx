import { FC } from "react"
import PageNewSchool from "./PageNewSchool"

export default {
  title: "Page/PageNewSchool",
  component: PageNewSchool,
  parameters: {
    componentSubtitle: "Just a simple PageNewSchool",
  },
}

export const Basic: FC = () => <PageNewSchool />
