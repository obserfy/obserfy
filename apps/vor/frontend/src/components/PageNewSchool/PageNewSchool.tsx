import { FC, FormEvent, useState } from "react"
import { Flex, Box, Button } from "theme-ui"
import { Trans } from "@lingui/macro"
import { navigate } from "../Link/Link"
import { Typography } from "../Typography/Typography"
import Input from "../Input/Input"

async function submitNewSchoolForm(name: string) {
  const data = { name }
  const response = await fetch("/api/v1/schools", {
    method: "POST",
    credentials: "same-origin",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  })
  if (response.status === 201) await navigate("/choose-school")
}

export const PageNewSchool: FC = () => {
  const [name, setName] = useState("")

  async function handleSubmit(e: FormEvent) {
    e.preventDefault()
    await submitNewSchoolForm(name)
  }

  return (
    <Flex
      sx={{
        justifyContent: "center",
        minHeight: "100vh",
        minWidth: "100vw",
      }}
      pt={6}
    >
      <Box
        as="form"
        p={3}
        sx={{ width: "100%", maxWidth: "maxWidth.sm" }}
        onSubmit={handleSubmit}
        mt={-5}
      >
        <Typography.H5 my={3}>New School</Typography.H5>
        <Input
          sx={{ width: "100%" }}
          label="School Name"
          value={name}
          onChange={(e) => setName(e.target.value)}
          required
          mb={3}
        />
        <Flex>
          <Button
            type="button"
            variant="outline"
            mr={3}
            onClick={() => navigate("/choose-school")}
            sx={{ flexGrow: 1 }}
          >
            <Trans>Cancel</Trans>
          </Button>
          <Button variant="primaryBig" sx={{ flexGrow: 1 }}>
            <Trans>Save</Trans>
          </Button>
        </Flex>
      </Box>
    </Flex>
  )
}

export default PageNewSchool
