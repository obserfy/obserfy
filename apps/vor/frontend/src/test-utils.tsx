/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
