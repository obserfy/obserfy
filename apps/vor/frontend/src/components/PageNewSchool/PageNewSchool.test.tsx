import { render } from "../../test-utils"
import { Basic } from "./PageNewSchool.stories"

describe("PageNewSchool", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
