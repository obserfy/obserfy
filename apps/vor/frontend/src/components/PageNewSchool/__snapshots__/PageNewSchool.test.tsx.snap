// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNewSchool should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 128px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-height: 100vh;
  min-width: 100vw;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  margin-top: -64px;
  width: 100%;
  max-width: 640px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-top: 16px;
  margin-bottom: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-7:autofill,
.emotion-7:autofill:hover,
.emotion-7:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-7:autofill:first-line,
.emotion-7:autofill:hover:first-line,
.emotion-7:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-7:-webkit-autofill,
.emotion-7:-webkit-autofill:hover,
.emotion-7:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-7:-webkit-autofill:first-line,
.emotion-7:-webkit-autofill:hover:first-line,
.emotion-7:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-7:hover,
.emotion-7:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-7::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-7::-moz-placeholder {
  font-style: italic;
}

.emotion-7:-ms-input-placeholder {
  font-style: italic;
}

.emotion-7::placeholder {
  font-style: italic;
}

.emotion-7:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-right: 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-9:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-9:hover,
.emotion-9:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-9:hover:disabled,
.emotion-9:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-10:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-10:hover,
.emotion-10:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-10:hover:disabled,
.emotion-10:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <form
        class="emotion-2"
      >
        <h5
          class="emotion-3"
        >
          New School
        </h5>
        <label
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            School Name
          </div>
          <div
            class="emotion-6"
          >
            <input
              class="emotion-7"
              required=""
              value=""
            />
          </div>
        </label>
        <div
          class="emotion-8"
        >
          <button
            class="emotion-9"
            type="button"
          >
            Cancel
          </button>
          <button
            class="emotion-10"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
